tests/test_resource_api.py

```diff
@@ -318,6 +316,154 @@ def test_resource_id(self):
             headers=admin_headers)
         self.assertEqual(resp.status_code, 404)
 
+    def test_get_member(self):
+        """Pages of resource ids are retrieved when using the get resource API"""
+        test_resource = 'test_get_member'
+        self.app.post(f'/v1/resource/{self.test_resource}/id/{test_resource}', headers=admin_headers)
+
+        # make groups
+        members = []
+        for group in [f'rt_group{i}' for i in range(5)]:
+            resp = self.app.post(
+                f'/v1/group',
+                data=json.dumps({'group_id': group}),
+                headers=admin_headers)
+            self.assertEqual(resp.status_code, 201)
+            members.append({'member': group,
+                            'member_type': 'group',
+                            'access_level': 'read'})
+
+        # make users
+        for user in [f'rt_user{i}' for i in range(6)]:
+            resp = self.app.post(
+                f'/v1/user',
+                data=json.dumps({'user_id': user}),
+                headers=admin_headers)
+            self.assertEqual(resp.status_code, 201)
+            members.append({'member': user,
+                            'member_type': 'user',
+                            'access_level': 'read'})
+        for m in members:
+            resp = self.app.put(
+                f'/v1/resource/{self.test_resource}/id/{test_resource}/members',
+                data=json.dumps([m]),
+                headers=admin_headers)
+            self.assertEqual(resp.status_code, 200)
+        self._test_paging(f'/v1/resource/{self.test_resource}/id/{test_resource}/members', admin_headers, 10, 'members')
+
+    def test_access_levels(self):
+        """Grant access to a principal for a resource"""
+        resource_id = 'protected-data'
+        arn_prefix = "arn:dcp:fus:us-east-1:dev:"
+        user = 'user_test_access_levels'
+        group = 'group_test_access_levels'
+
+        # create the resource to control
+        resp = self.app.post(
+            f'/v1/resource/{self.test_resource}/id/{resource_id}',
+            headers=admin_headers)
+        self.assertEqual(resp.status_code, 201)
+
+        with self.subTest("Check that no one has access by listing who has access"):
+            resp = self.app.get(
+                f'/v1/resource/{self.test_resource}/id/{resource_id}/members',
+                headers=admin_headers)
+            self.assertEqual(resp.status_code, 200)
+            self.assertJSONEqual(resp.body, {'members': []})
```

Comment:
if the API gets changed to return more information you'll have to refactor this. no biggie tho.

Reply:
I can live with that